src/com/android/launcher3/popup/SystemShortcut.java

```diff
@@ -379,6 +382,36 @@ public void onClick(View view) {
                     .log(LAUNCHER_PRIVATE_SPACE_UNINSTALL_SYSTEM_SHORTCUT_TAP);
         }
     }
+    
+    public static final Factory<ActivityContext> FREE_FORM = (activity, itemInfo, originalView) ->
+        new FreeForm(activity, itemInfo, originalView);
+    public static class FreeForm<T extends ActivityContext> extends SystemShortcut<T> {
+        private final String mPackageName;
+        private final ComponentName mComponentName;
+        private final int mUserId;
+
+        public FreeForm(T target, ItemInfo itemInfo, View originalView) {
+            super(R.drawable.ic_caption_desktop_button_foreground, R.string.recent_task_option_freeform, target, itemInfo, originalView);
+            mPackageName = itemInfo.getTargetComponent().getPackageName();
+            mComponentName = itemInfo.getTargetComponent();
+            mUserId = originalView.getContext().getUserId();
+        }
+        @Override
+        public void onClick(View view) {
+            if (mPackageName != null) {
+                startLmoFreeform(view.getContext());
+                AbstractFloatingView.closeAllOpenViews(((ActivityContext) mTarget));
+            }
+        }
+        private void startLmoFreeform(Context context) {
+            final Intent intent = new Intent("com.libremobileos.freeform.START_FREEFORM")
+                .setPackage("com.libremobileos.freeform")
+                .putExtra("packageName", mPackageName)
+                .putExtra("activityName", mComponentName.getClassName())
+                .putExtra("userId", mUserId);
+            context.sendBroadcast(intent);
+        }
+    }
```

Comment:
Please move this to [LawnchairShortcut.kt](https://github.com/LawnchairLauncher/lawnchair/blob/15-dev/lawnchair/src/app/lawnchair/ui/popup/LawnchairShortcut.kt) 

the usage can be found [here](https://github.com/LawnchairLauncher/lawnchair/blob/15-dev/lawnchair/src/app/lawnchair/LawnchairLauncher.kt#L231)

Reply:
Ehmmmm... I have no idea how to code.